src/Extractor/TranslationsExtractor.php

```diff
@@ -0,0 +1,134 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\Translations\Extractor;
+
+use Becklyn\Translations\Cache\CacheDigestGenerator;
+use Becklyn\Translations\Catalogue\CachedCatalogue;
+use Becklyn\Translations\Catalogue\KeyCatalogue;
+use Symfony\Component\HttpFoundation\File\Exception\UnexpectedTypeException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class TranslationsExtractor
+{
+    private const CACHE_PREFIX = "becklyn_translations.catalogue";
+
+
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+
+    /**
+     * @var CacheDigestGenerator
+     */
+    private $cacheDigestGenerator;
+
+
+    /**
+     * @var KeyCatalogue
+     */
+    private $catalogue;
+
+
+    /**
+     * @param CacheInterface       $cache
+     * @param TranslatorInterface  $translator
+     * @param CacheDigestGenerator $cacheDigestGenerator
+     * @param KeyCatalogue         $catalogue
+     */
+    public function __construct (
+        CacheInterface $cache,
+        TranslatorInterface $translator,
+        CacheDigestGenerator $cacheDigestGenerator,
+        KeyCatalogue $catalogue
+    )
+    {
+        $this->cache = $cache;
+        $this->translator = $translator;
+        $this->cacheDigestGenerator = $cacheDigestGenerator;
+        $this->catalogue = $catalogue;
+    }
+
+
+    /**
+     * Fetches the catalogue for the given language.
+     *
+     * @param string $locale
+     * @param bool   $useCache
+     *
+     * @return CachedCatalogue
+     */
+    public function fetchCatalogue (string $locale, bool $useCache = true) : CachedCatalogue
+    {
+        $fetchCallback = function () use ($locale)
+        {
+            $catalogueJson = \json_encode(
+                $this->extractCatalogue($locale),
+                JsonResponse::DEFAULT_ENCODING_OPTIONS | \JSON_THROW_ON_ERROR
+            );
+
+            return new CachedCatalogue(
+                $this->cacheDigestGenerator->calculateDigest($catalogueJson),
+                $catalogueJson
+            );
+        };
+
+        return $useCache
+            ? $this->cache->get(self::CACHE_PREFIX . ".{$locale}", $fetchCallback)
+            : $fetchCallback();
```

Comment:
Do we need to catch the exception to re-throw a custom exception?

Reply:
We could validate the domain, but otherwise the exception can’t occur.

Reply:
I added the validation on the route - this is the only way to this method.